### PR TITLE
Makefile: Add a generate-changelog target

### PR DESCRIPTION
**Pull request by timflannagan**
# Description

This commit introduces a new `generate-changelog` target that makes it easy to
generate changelog entry stubs. The new target calls the existing bash script
under-the-hood and mirrors the way that target was implemented in the GME
repository. This helps improve devexp as we're interfacing with make targets
directly (vs. calling bash scripts) and the target shows up the the `help` output.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -421,6 +421,10 @@ generated-code-apis: clean-solo-kit-gen go-generate-apis fmt ## Executes the tar
 .PHONY: generated-code-cleanup
 generated-code-cleanup: getter-check mod-tidy update-licenses fmt ## Executes the targets necessary to cleanup and format code
 
+.PHONY: generate-changelog ## Generate a changelog entry
+generate-changelog:
+	@./ci/changelog.sh
+
```
Comment (timflannagan) [inline Makefile:427]: I could move this to it's own section:

```make
#----------------------------------------------------------------------------------
# Generate <something>
#----------------------------------------------------------------------------------
```

But I felt like that was overkill.

File: Makefile [hunk 2]
```diff
@@ -421,6 +421,10 @@ generated-code-apis: clean-solo-kit-gen go-generate-apis fmt ## Executes the tar
 .PHONY: generated-code-cleanup
 generated-code-cleanup: getter-check mod-tidy update-licenses fmt ## Executes the targets necessary to cleanup and format code
 
+.PHONY: generate-changelog
+generate-changelog: ## Generate a changelog entry
+	@./devel/tools/changelog.sh
+
```
Comment (timflannagan) [inline Makefile:427]: See https://solo-io-corp.slack.com/archives/C06C8RA01NF/p1724096357041519 for some context. Jacob mentioned that the changelog.sh bash script supports a single "edit" parameter, so we could define a CHANGELOG_ARGS-esq variable (e.g. `make generate-changelog CHANGELOG_ARGS=--edit`) that gets passed to the script, but that felt like overkill as the script doesn't support much user input right now.

**Review by sam-heilbron — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -421,6 +421,10 @@ generated-code-apis: clean-solo-kit-gen go-generate-apis fmt ## Executes the tar
 .PHONY: generated-code-cleanup
 generated-code-cleanup: getter-check mod-tidy update-licenses fmt ## Executes the targets necessary to cleanup and format code
 
+.PHONY: generate-changelog
+generate-changelog: ## Generate a changelog entry
+	@./devel/tools/changelog.sh
```
Comment (jbohanon) [inline Makefile:426]:
```suggestion
	@./devel/tools/changelog.sh edit
```

**Review by jbohanon — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (timflannagan) [reply Makefile:427]: Let's add that as a follow-up if folks feel like it's the better UX.

**Pull request merged**